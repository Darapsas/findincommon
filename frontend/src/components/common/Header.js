import React, { Component } from "react";
import { Link, NavLink } from "react-router-dom";
import "./Header.css";

export default props => (
  <header className="app-header">
    <div className="container">
      <div className="app-branding">
        <Link to="/" className="app-title">
          Find In Common
        </Link>
      </div>
      <div className="app-options">
        <nav className="app-nav">
          {props.authenticated ? (
            <ul>
              <li>
                <NavLink to="/myHobbies">Hobbies</NavLink>
              </li>
              <li>
                <NavLink to="/myEvents">Events</NavLink>
              </li>
              <li>
                <NavLink to="/myConversations">Conversations</NavLink>
              </li>
              <li>
                <NavLink to="/profile">Profile</NavLink>
              </li>
              <li>
                <a onClick={props.onLogout}>Logout</a>
              </li>
            </ul>
          ) : (
            <ul>
              <li>
                <NavLink to="/signin">Sign in</NavLink>
              </li>
            </ul>
          )}
        </nav>
      </div>
    </div>
  </header>
);
