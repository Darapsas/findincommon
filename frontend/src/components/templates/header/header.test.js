import React from 'react'
import Header from './header'
import { shallow, mount } from 'enzyme'
import renderer from 'react-test-renderer'
import { MemoryRouter, Link } from 'react-router-dom'

describe('<Header />', () => {
  it('matches the snapshot', () => {
    const tree = renderer
      .create(
        <MemoryRouter>
          <Header />
        </MemoryRouter>
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('check something', () => {
    const wrapper = mount(
      <MemoryRouter>
        <Header />
      </MemoryRouter>
    )
    const linkButton = wrapper.find('button')
    expect(linkButton).toBeDefined()
  })
})
