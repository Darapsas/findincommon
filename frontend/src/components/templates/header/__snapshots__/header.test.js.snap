// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> matches the snapshot 1`] = `
<header>
  <nav
    className="navbar navbar-expand-md navbar-dark fixed-top bg-dark"
  >
    <a
      className="navbar-brand app-title"
      href="/"
      onClick={[Function]}
    >
      Find In Common
    </a>
    <button
      aria-controls="navbarCollapse"
      aria-expanded="false"
      aria-label="Toggle navigation"
      className="navbar-toggler"
      data-target="#navbarCollapse"
      data-toggle="collapse"
      type="button"
    >
      <span
        className="navbar-toggler-icon"
      />
    </button>
    <div
      className="collapse navbar-collapse"
      id="navbarCollapse"
    >
      <ul
        className="navbar-nav ml-auto"
      >
        <li
          className="flaot-right"
        >
          <a
            aria-current={null}
            className="nav-link"
            href="/sign-in"
            onClick={[Function]}
          >
            Sign in
          </a>
        </li>
      </ul>
    </div>
  </nav>
</header>
`;
